amundsen_application/dashboard_preview/preview_factory_method.py

```diff
@@ -0,0 +1,40 @@
+import logging
+from abc import ABCMeta, abstractmethod
+
+from amundsen_application.dashboard_preview.base_preview import BasePreview
+from amundsen_application.dashboard_preview.mode_preview import ModePreview
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BasePreviewMethodFactory(metaclass=ABCMeta):
+
+    @abstractmethod
+    def get_instance(self, *, uri: str) -> BasePreview:
+        """
+        Provides an instance of BasePreview based on uri
+        :param uri:
+        :return:
+        """
+        pass
+
+
+class DefaultPreviewMethodFactory(BasePreviewMethodFactory):
```

Comment:
This is a very useful idea. In the future so you think we should do something similar with table preview clients  and all other custom clients? Right now I think our frontend configurations are getting trickier for people to figure out on their own because we have to many different approaches:
1. We used entry point for table preview client and announcement client
2. We used the flask config.py for the mail client
3. Now we have this approach for dashboard preview client.

Appreciate any opinions on what you think would be best moving forward and how we can eventually make things easier to figure out for the community.

Reply:
> 3\. Now we have this approach for dashboard preview client.

Yeah, I think `3` is more for a way to inject implementation when there's different implementation needs to be used based on data. For example, we can add more support on Dashboard as we start supporting Tableau and Superset. `entry point` provides similar structure, but the downside is that not many people is familiar with it.

I think `2` approach is still valid where it does not need a factory pattern -- for the use case where just one implementation is enough.

Thus, I think we can go with approach `2` and `3` going forward.